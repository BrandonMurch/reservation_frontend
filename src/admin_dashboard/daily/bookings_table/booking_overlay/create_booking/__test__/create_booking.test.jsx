// Dependencies
import React from 'react';
import {
  render, fireEvent, screen,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { unmountComponentAtNode } from 'react-dom';
import { create } from 'react-test-renderer';

// Components
import CreateBooking from '../index';

describe('<CreateBooking />', () => {
  let container = null;
  const props = {
    date: '2020-10-24',
    onSubmit: jest.fn(),
  };

  beforeEach(async () => {
    container = document.createElement('div');
    document.body.appendChild(container);
    render(
      <CreateBooking {...props} />,
    );
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('should match snapshot', () => {
    const tree = create(
      <CreateBooking {...props} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onSubmit on submit', async () => {
    //
    fireEvent.change(screen.getByLabelText(/start time/i), { target: { value: '21:00' } });
    fireEvent.change(screen.getByLabelText(/end time/i), { target: { value: '23:00' } });
    userEvent.type(screen.getByLabelText(/party size/i), '2');
    userEvent.type(screen.getByLabelText(/name/i), 'John');
    userEvent.type(screen.getByLabelText(/phone number/i), '+1 1234566');
    userEvent.type(screen.getByLabelText(/email/i), 'john@john.com');
    userEvent.type(screen.getByLabelText(/restaurant comments/i), 'comment');
    userEvent.click(screen.getByRole('button', { name: /create booking/i }));
    expect(props.onSubmit).toHaveBeenCalled();
  });

  it('should display date as default', () => {
    const dateInput = screen.getByLabelText(/date/i);
    expect(dateInput).toHaveValue(props.date);
  });
});
