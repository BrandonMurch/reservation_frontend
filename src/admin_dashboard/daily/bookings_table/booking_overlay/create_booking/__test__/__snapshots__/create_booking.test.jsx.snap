// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateBooking /> should match snapshot 1`] = `
<div
  className="formContainer"
>
  <form
    className="container"
    data-testid="form"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="date"
      >
        Date
      </label>
      <input
        className="input"
        id="date"
        name="date"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        pattern={null}
        placeholder="Date"
        required={true}
        type="date"
        value="2020-10-24"
      />
    </div>
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="startTime"
      >
        Start Time
      </label>
      <input
        className="input"
        id="startTime"
        name="startTime"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        pattern={null}
        placeholder="Start Time"
        required={true}
        type="time"
        value=""
      />
    </div>
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="endTime"
      >
        End Time
      </label>
      <input
        className="input"
        id="endTime"
        name="endTime"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        pattern={null}
        placeholder="End Time"
        required={false}
        type="time"
        value=""
      />
    </div>
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="partySize"
      >
        Party Size
      </label>
      <input
        className="input"
        id="partySize"
        name="partySize"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        pattern="^[1-9][0-9]*$"
        placeholder="Party Size"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="input"
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        pattern={null}
        placeholder="Name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="input"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        pattern="^[\\\\w\\\\-_.+]*[\\\\w\\\\-_.]@([\\\\w]+\\\\.)+[\\\\w]+[\\\\w]$"
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="phoneNumber"
      >
        Phone Number
      </label>
      <input
        className="input"
        id="phoneNumber"
        name="phoneNumber"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        pattern="^\\\\+\\\\d{1,3} \\\\d{6,14}$"
        placeholder="Phone Number"
        required={true}
        type="tel"
        value=""
      />
    </div>
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="restaurantComments"
      >
        Restaurant Comments
      </label>
      <input
        className="input"
        id="restaurantComments"
        name="restaurantComments"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        pattern={null}
        placeholder="Restaurant Comments"
        required={false}
        type="text-area"
        value=""
      />
    </div>
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="userComments"
      >
        User Comments
      </label>
      <input
        className="input"
        id="userComments"
        name="userComments"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        pattern={null}
        placeholder="User Comments"
        required={false}
        type="text-area"
        value=""
      />
    </div>
    <input
      className="button"
      disabled={false}
      type="submit"
      value="Create booking"
    />
  </form>
</div>
`;
